app/controllers/areas_controller.rb

```diff
@@ -1,9 +1,12 @@
 class AreasController < ApplicationController
-  before_action :set_city, only: [:edit, :new, :create, :update]
+  before_filter :authenticate_user!
+  load_and_authorize_resource
+  before_action :set_city, only: [:index,:edit, :new, :create, :update]
   before_action :set_area, only: [:show, :edit, :update, :destroy]
-
+  before_action :set_city
   def index
-    @areas = Area.all
+    @areas = @city.areas.all.order_by(:name => 'asc')
+    @area = Area.new
```

Comment:
@shaunak-amura  because I am adding cities via ajax call where create city and view all city is on same page